railties/lib/rails/application.rb

```diff
@@ -469,10 +469,13 @@ def credentials
     #
     #   Rails.application.encrypted("config/special_tokens.yml.enc", key_path: "config/special_tokens.key")
     def encrypted(path, key_path: "config/master.key", env_key: "RAILS_MASTER_KEY")
+      require_master_key = config.respond_to?(:require_master_key) && config.require_master_key
+
       ActiveSupport::EncryptedConfiguration.new(
         config_path: Rails.root.join(path),
         key_path: Rails.root.join(key_path),
-        env_key: env_key
+        env_key: env_key,
+        raise_if_missing_key: require_master_key
```

Comment:
Think `config[:require_master_key]` could spare us the `respond_to?` check.

Reply:
Unfortunately, `Rails::Application::Configuration` does not support `[]` method for now.

```ruby
config[:require_master_key]
#=> NoMethodError Exception: undefined method `[]' for #<Rails::Application::Configuration:0x005600b330a840 
```

If it says to add a method, I agree!

Reply:
Oh! I thought it was an instance of `ActiveSupport::OrderedOptions`. Nevermind then.

Reply:
No wait, we should just add the `attr_accessor`.

Reply:
Oh, :+1: